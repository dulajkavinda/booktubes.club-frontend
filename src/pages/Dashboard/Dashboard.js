import React from "react";
import Navbar from "../../containers/NavBar/Navbar";
import { Button } from "@chakra-ui/react";
import Club from "../../components/Club/Club";
import { Divider } from "@chakra-ui/react";
import styles from "./Dashboard.module.css";
import Link from "next/link";
import { useStoreState } from "easy-peasy";

import clubdata from "../../data/clubs.json";

export default function Dashboard() {
  const user = useStoreState((state) => state.user);

  return (
    <div>
      <Navbar username={user.displayName} />
      <div className={styles.main}>
        <div className={styles.create}>
          <span className={styles.title}>My Clubs</span>

          <Button ml={5} variant={"outline"} mr={5} colorScheme="blue">
            <Link href={"/create/AddClub"}>
              <a> Create Your Own Club</a>
            </Link>
          </Button>
        </div>

        <Divider />
        <div className={styles.myclub}>
          {clubdata.data.map((club) => {
            if (club.members.includes(user.uid)) {
              return <Club type="my" club={club} />;
            }
          })}
        </div>
        <span className={styles.title}>Explore More Clubs</span>
        <Divider />
        <div className={styles.myclub}>
          {clubdata.data.map((club) => {
            return <Club type="all" club={club} />;
          })}
        </div>
      </div>
    </div>
  );
}
