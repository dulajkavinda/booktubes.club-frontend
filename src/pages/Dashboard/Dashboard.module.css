.myclub {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  width: 100%;
  height: 400px;
  flex-wrap: wrap;
  flex-direction: column;
  overflow-x: auto;
}

.main {
  display: flex;
  justify-content: center;
  align-items: center;

  flex-direction: column;
}

.title {
  color: #1a1c26;
  opacity: 0.6;
  font-size: 1.2rem;
  font-weight: 600;
  letter-spacing: -0.5px;
  display: -webkit-box;
  -webkit-line-clamp: 4; /* number of lines to show */
  -webkit-box-orient: vertical;
  margin-top: 10px;
  font-weight: 700;

  margin-top: 10px;
  margin-bottom: 10px;
}

.create {
  display: flex;
  justify-content: center;
  align-items: center;
  text-align: center;

  margin-top: 10px;
  margin-bottom: 10px;
}
