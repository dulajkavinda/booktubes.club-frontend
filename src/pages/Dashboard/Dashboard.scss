@import "../../styles/config";

.dashboard_main {
  margin: 10px;
  .dashboard_title {
    display: flex;
    align-items: center;
    height: 100px;

    font-family: $primary-text-style;
    font-weight: 800;
    font-size: $primary-text-size;
    color: $secondary-text-color;
  }

  .dashboard_reading {
    @extend .dashboard_title;
    font-size: $secondary-text-size;
    color: $sub-text-color;
    height: 50px;
  }
}

.dashboard_list {
  @include flexCenter(row);
  justify-content: flex-start;
}
