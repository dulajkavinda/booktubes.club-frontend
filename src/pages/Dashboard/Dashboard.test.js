import React from "react";
import { shallow, mount } from "enzyme";
import Dashboard from "./Dashboard";

let wrapper;
beforeEach(() => {
  wrapper = mount(<Dashboard />);
});

describe("dashboard basic tests", () => {
  it("is dashboard renders", () => {
    expect(wrapper.find(".dashboard-title").text()).toEqual("Dashboard");
  });
});
