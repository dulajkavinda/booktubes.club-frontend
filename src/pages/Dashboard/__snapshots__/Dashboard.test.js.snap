// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render self and subcomponents 1`] = `
<Dashboard>
  <div>
    <p
      id="dash"
    >
      Dashboard
    </p>
  </div>
</Dashboard>
`;
